src/Adapters/MockAdapter.ts

```diff
@@ -719,71 +717,106 @@ export default class MockAdapter
+            return new AdapterResult({
+                result: new AzureResourcesData(mockStorageContainerResources),
+                errorInfo: null
+            });
+        } else if (resourceType === AzureResourceTypes.StorageAccount) {
             return new AdapterResult({
-                result: new AzureResourcesData(mockContainerResources),
+                result: new AzureResourcesData(mockStorageAccountResources),
                 errorInfo: null
             });
         } else {
             return new AdapterResult({
-                result: null,
+                result: new AzureResourcesData([]),
                 errorInfo: null
             });
         }
     }
 
-    async getADTInstances() {
+    async getResourcesByPermissions(
+        resourceType: AzureResourceTypes,
+        _requiredAccessRoles: {
+            enforcedRoleIds: Array<AzureAccessPermissionRoles>; // roles that have to exist
+            interchangeableRoleIds: Array<AzureAccessPermissionRoles>; // roles that one or the other has to exist
+        },
+        searchParams?: {
+            take?: number;
+            filter?: string;
+            additionalParams?: {
+                storageAccountId?: string;
+                [key: string]: any;
+            };
+        }
+    ) {
         try {
-            const adtInstanceResourcesResult = await this.getResources(
-                AzureResourceTypes.ADT,
-                AzureResourceProviderEndpoints.ADT
-            );
-            const adtInstanceResources: Array<IAzureResource> = adtInstanceResourcesResult.getData();
-            const digitalTwinsInstances: Array<IADTInstance> = adtInstanceResources.map(
-                (adtInstanceResource) =>
-                    ({
-                        id: adtInstanceResource.id,
-                        name: adtInstanceResource.name,
-                        hostName: adtInstanceResource.properties['hostName'],
-                        location: adtInstanceResource.location
-                    } as IADTInstance)
-            );
+            const getResourcesResult = await this.getResources(resourceType);
+            let resources: Array<IAzureResource> = getResourcesResult.getData();
+
+            if (resources?.length) {
+                // apply searchParams to the list of resources returned
+                if (searchParams?.filter) {
+                    resources = resources.filter((resource) =>
+                        Object.keys(AzureResourceDisplayFields).some(
+                            (displayField) =>
+                                !!resource[displayField]?.includes(displayField)
+                        )
+                    );
+                }
+                resources = resources.slice(
+                    0,
+                    searchParams?.take || MAX_RESOURCE_TAKE_LIMIT
+                ); // take the first n number of resources to make sure the browser won't crash with making thousands of requests
```

Comment:
it was for preventing the browser to crash with thousands of calls, by taking a max number of resolved resources right before making another batch of calls for permission check for each resource. But since we use search params for containers based on storage account id, we may no longer need that. (for example, I have ~700 storage account and the whole getResources method takes around 60 seconds with around 1200 requests). I am going to remove take now but will keep it in comment for future use, users probably would have much more less than those number of resources.